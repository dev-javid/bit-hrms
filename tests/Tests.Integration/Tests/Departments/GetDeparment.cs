namespace Tests.Integration.Tests.Departments
{
    public class GetDeparment(TestWebApplicationFactory<Program> factory) : IntegrationTest(factory)
    {
        private const string Route = "/api/departments/1";

        [Fact]
        [UseReporter(typeof(DiffReporter))]
        public async Task Should_Return_Data()
        {
            // Arrange
            await LoginAsCompanyAdminAsync();
            await FeedDataAsync("Tests/Departments/GetDeparment.sql");

            // Act
            using HttpResponseMessage response = await Client.GetAsync(Route);

            // Assert
            response.StatusCode.Should().Be(HttpStatusCode.OK);
            var result = await response.Content.ReadAsStringAsync();
            Approvals.VerifyJson(result);
        }
    }
}
