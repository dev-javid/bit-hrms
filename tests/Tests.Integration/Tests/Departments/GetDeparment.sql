INSERT INTO public.departments (id, company_id, name, created_by, created_on)
VALUES
(1, 999, 'HR',          1, now()),
(2, 999, 'Finance',     1, now()),
(3, 999, 'Engineering', 1, now()),
(4, 998, 'Marketing',   1, now());


INSERT INTO public.job_titles (id, company_id,  department_id, name, created_by, created_on)
VALUES
(1, 999, 1, 'Teacher',  1, now())
