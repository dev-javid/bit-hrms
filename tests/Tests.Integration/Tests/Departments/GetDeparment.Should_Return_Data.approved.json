{
    "departmentId": 1,
    "name": "HR",
    "jobTitles": [
        {
            "name": "Teacher"
        }
    ]
}